tests/unittest/test_tokens.py

```diff
@@ -38,6 +41,48 @@ def test__should_encode_and_decode_correctly(self):
 
         # -- Assert ----------------------------------------------------------
 
-        assert isinstance(decoded, Message1)
+        assert isinstance(decoded, MockMessage)
         assert decoded != encoded
         assert decoded == obj
+
+    def test__tamper_with_token_payload__should_fail_to_decode(self):
```

Comment:
This test decodes and re-encodes with an invalid key.
I would suggest a test where the content of the JWT is modified without reencoding it, but changing the body with the existing signature.

Reply:
I have added a test which tests for this exact thing.